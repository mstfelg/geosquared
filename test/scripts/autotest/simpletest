#!/bin/sh
# Checks if the current build works properly (basic checks)

# 1. Start and loading configuration
test -r autotest.conf || {
 echo "$0: Cannot find autotest.conf (make sure you edit autotest.conf.dist first)"
 exit 127
 }
test -r autotest.conf && . ./autotest.conf

MYDIR=$SVNDIR/geogebra/test/scripts/autotest

REVISION=`./myrevision`

cd $REVDIR/$REVISION || exit 1 # Build problem

test -r geogebra.jar || exit 2 # Non-existing JAR

# Testing no-3D version:
xvfb-run --auto-servernum timeout -k $SIMPLETESTTIMEOUT $SIMPLETESTTIMEOUT \
 $JAVA_HOME/bin/java -cp geogebra.jar geogebra.GeoGebra3D \
  --regressionFile=simpletest.out --logFile=simpletest.log $MYDIR/simpletest.ggb \
  >simpletest.stdout 2>simpletest.stderr

diff simpletest.out $MYDIR/simpletest.out || exit 3 # Problem with simple testing

ECJERROR=`cat ecjerrors`
if [ "$ECJERROR" != "" ]; then
 exit 4 # Compilation error (but GeoGebra started)
 fi

WEBERROR=`cat weberrors`
if [ "$WEBERROR" != "" ]; then
 exit 5 # Compilation error (but GeoGebra started)
 fi

exit 0 # No error
