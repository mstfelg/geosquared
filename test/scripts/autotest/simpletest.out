Point A = Intersect[xAxis, yAxis] == 0.0 -0.0
Point B = 1 1 1
Line a = Line[A, B] == -1 1 0.0
